staging/src/k8s.io/apiserver/pkg/audit/request.go

```diff
@@ -244,3 +259,62 @@ func maybeTruncateUserAgent(req *http.Request) string {
 
 	return ua
 }
+
+// copyWithoutManagedFields will make a deep copy of the specified object and will discard
+// managed fields from the copy.
+// The specified object is expected to be a meta.Object or a "list". The specified object obj
+// is treated as readonly and hence not mutated.
+// On return, an error is set if the function runs into any error while removing the managed fields.
+// On return, the boolean value is true if the copy has been made successfully, otherwise false.
+func copyWithoutManagedFields(obj runtime.Object) (runtime.Object, bool, error) {
+	removeManagedFields := func(obj runtime.Object) error {
+		accessor, err := meta.Accessor(obj)
+		if err != nil {
+			return err
+		}
+		accessor.SetManagedFields(nil)
+		return nil
+	}
+
+	isAccessor := true
+	if _, err := meta.Accessor(obj); err != nil {
+		isAccessor = false
+	}
+	isList := meta.IsListType(obj)
+
+	_, isTable := obj.(*metav1.Table)
+	if !isAccessor && !isList && !isTable {
+		return nil, false, nil
+	}
+
+	copy := obj.DeepCopyObject()
```

Comment:
Hi @wojtek-t, the deep copy is happening:
- only when the user is logging at `LevelRequest` and `LevelRequestResponse` level
- the deep copy is done only when it's necessary.

I would like your thoughts on the deep copy here, I understand that a deep copy can be expensive.

Reply:
This is once per-request, we are doing more copies/serializations/deserializations/etc. for a given request. That sounds fine to me (I mean, if we can avoid that somehow, let's avoid, but I don't see an easy way for it).

Reply:
Why would this require a deep copy? it only exists for long enough to be serialized?

Reply:
As mentioned above - I don't think it has to be a deep-copy, but I don't know how to generically make a shallow copy here. I requested a TODO above.

Reply:
yes, I have added a TODO

Reply:
do we know the relative cost (cpu/allocs) of adding a deep copy into the audit pipeline by enabling the omitManagedFields option globally?

Reply:
I think it would be similar to the other deep copies we have in place today. I have not done any benchmarking to measure it. I opened this issue so we can track it - https://github.com/kubernetes/kubernetes/issues/106094